src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidationResult.cs

```diff
@@ -1,112 +1,167 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#nullable enable
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Security.Claims;
+using System.Threading;
 
-#nullable enable
 namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Contains results of a single step in validating a <see cref="SecurityToken"/>.
     /// A <see cref="TokenValidationResult"/> maintains a list of <see cref="ValidationResult"/> for each step in the token validation.
     /// </summary>
-    internal abstract class ValidationResult
+    internal class ValidationResult
     {
-        private bool _isValid;
-
         /// <summary>
         /// Creates an instance of <see cref="ValidationResult"/>
         /// </summary>
-        protected ValidationResult()
+        /// <param name="securityToken">The <see cref="SecurityToken"/> that is being validated.</param>
+        /// <param name="tokenHandler">The <see cref="TokenHandler"/> that is being used to validate the token.</param>
+        /// <param name="validationParameters">The <see cref="ValidationParameters"/> to be used for validating the token.</param>
+        internal ValidationResult(
+            SecurityToken securityToken,
+            TokenHandler tokenHandler,
+            ValidationParameters validationParameters)
         {
-            ValidationFailureType = ValidationFailureType.ValidationNotEvaluated;
+            TokenHandler = tokenHandler ?? throw new ArgumentNullException("TokenHandler cannot be null.");
```

Comment:
when ArgumentNullException has a single string param, it is supposed to be the parameter name.